mythril/mythril.py

```diff
@@ -481,23 +481,31 @@ def fire_lasers(
 
         all_issues = []
         for contract in contracts or self.contracts:
-            sym = SymExecWrapper(
-                contract,
-                address,
-                strategy,
-                dynloader=DynLoader(
-                    self.eth,
-                    storage_loading=self.onchain_storage_access,
-                    contract_loading=self.dynld,
-                ),
-                max_depth=max_depth,
-                execution_timeout=execution_timeout,
-                create_timeout=create_timeout,
-                transaction_count=transaction_count,
-                modules=modules,
-            )
-
-            issues = fire_lasers(sym, modules)
+            try:
+                sym = SymExecWrapper(
+                    contract,
+                    address,
+                    strategy,
+                    dynloader=DynLoader(
+                        self.eth,
+                        storage_loading=self.onchain_storage_access,
+                        contract_loading=self.dynld,
+                    ),
+                    max_depth=max_depth,
+                    execution_timeout=execution_timeout,
+                    create_timeout=create_timeout,
+                    transaction_count=transaction_count,
+                    modules=modules,
+                )
+                issues = fire_lasers(sym, modules)
+            except KeyboardInterrupt:
+                log.critical("Keyboard Interrupt")
+                issues = retrieve_callback_issues(modules)
+            except Exception:
+                log.critical(
+                    "Exception occurred, aborting analysis\n" + traceback.format_exc()
+                )
+                issues = retrieve_callback_issues(modules)
```

Comment:
Issues are gathered differently depending on whether or not there was an exception. `finally` executes whether or not an exception is thrown, so I'm not sure it would be appropriate in this situation. This StackOverflow question makes me think it would be nicer just to leave the one extra line in: <https://stackoverflow.com/questions/34796808/share-python-code-when-handling-multiple-exceptions>.

Reply:
Okay, lets just keep it in there then